qiskit/transpiler/passes/mapping/basic_mapper.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+A pass implementing a basic mapper.
+
+The basic mapper is a minimum effort to insert swap gates to map the DAG into a coupling map. When
+a cx is not in the coupling map possibilities, it inserts one or more swaps in front to make it
+compatible.
+"""
+
+from copy import copy
+
+from qiskit.transpiler._basepasses import TransformationPass
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.mapper import Layout
+from qiskit.extensions.standard import SwapGate
+
+
+class BasicMapper(TransformationPass):
+    """
+    Maps (with minimum effort) a DAGCircuit onto a `coupling_map` adding swap gates.
+    """
+
+    def __init__(self,
+                 coupling_map,
+                 swap_gate=None,
```

Comment:
Can you get rid of this argument? There is no reason to pass the `swap_gate`. A swap is always a swap.